pkg/model/iam/iam_builder.go

```diff
@@ -49,12 +49,35 @@ const PolicyDefaultVersion = "2012-10-17"
 
 // Policy Struct is a collection of fields that form a valid AWS policy document
 type Policy struct {
-	Statement []*Statement
-	Version   string
+	clusterName         string
+	unconditionalAction sets.String
+	clusterTaggedAction sets.String
+	Statement           []*Statement
+	Version             string
```

Comment:
Does this need to be public? We could have separate builder and built types. (for future PR)

Reply:
Yeah, I am not happy about this, but this is due to consumers calling `toJSON` on the built type directly. Its a bit more work to separate this out so that they rather receive a string resource or something.